scripts/west_commands/runners/stlink_gdbserver.py

```diff
@@ -0,0 +1,188 @@
+Runner for debugging applications using the ST-LINK GDB server
+from STMicroelectronics, provided as part of the STM32CubeCLT.
+"""
+
+import argparse
+import platform
+import re
+import shutil
+from pathlib import Path
+
+from runners.core import RunnerCaps, RunnerConfig, ZephyrBinaryRunner
+
+STLINK_GDB_SERVER_DEFAULT_PORT = 61234
+
+
+class STLinkGDBServerRunner(ZephyrBinaryRunner):
+    @classmethod
+    def _get_stm32cubeclt_paths(cls) -> tuple[Path, Path]:
+        """
+        Returns a tuple of two elements of class pathlib.Path:
+            [0]: path to the ST-LINK_gdbserver executable
+            [1]: path to the "STM32CubeProgrammer/bin" folder
+        """
+
+        def find_highest_clt_version(tools_folder: Path) -> Path | None:
+            CUBECLT_FLDR_RE = re.compile("stm32cubeclt_([1-9]).([0-9]+).([0-9]+)", re.IGNORECASE)
+
+            def path2version(path: Path):
+                m = CUBECLT_FLDR_RE.match(path.name)
+                if m is None:
+                    # We should never be calling this function for
+                    # a path that did not match() the regexp, so
+                    # this should never happen...
+                    raise RuntimeError("Invalid candidate path?!")
+
+                major = int(m[1])
+                minor = int(m[2])
+                revis = int(m[3])
+                return major * 1000000 + minor * 1000 + revis
+
+            # List all possible CubeCLT installations in tools folder
+            print([k.name for k in tools_folder.iterdir()])
+
+            candidates = [f for f in tools_folder.iterdir() if CUBECLT_FLDR_RE.match(f.name)]
+
+            if len(candidates) == 0:
+                return None
+
+            # Sort candidates and return the most recent version
+            return next(reversed(sorted(candidates, key=path2version)))
+
+        cur_platform = platform.system()
+
+        # Attempt to find via shutil.which()
+        if cur_platform in ["Linux", "Windows"]:
+            gdbserv = shutil.which("ST-LINK_gdbserver")
+            cubeprg = shutil.which("STM32_Programmer_CLI")
+            if gdbserv and cubeprg:
+                # Return the parent of cubeprg as [1] should be the path
+                # to the folder containing STM32_Programmer_CLI, not the
+                # path to the executable itself
+                return (Path(gdbserv), Path(cubeprg).parent)
+
+        # Search in OS-specific paths
+        search_path: str
+        tool_suffix = ""
+        if cur_platform == "Linux":
+            search_path = "/opt/st/"
+        elif cur_platform == "Windows":
+            search_path = "C:\\ST\\"
+            tool_suffix = ".exe"
+        elif cur_platform == "Darwin":
+            search_path = "/opt/ST/"
+        else:
+            raise NotImplementedError("Unsupported OS")
+
+        clt = find_highest_clt_version(Path(search_path))
+        if clt is None:
+            raise RuntimeError("Could not locate STM32CubeCLT installation!")
```

Comment:
Ditto here, and all next uses of `raise`.

```suggestion
            self.logger.error("Could not locate STM32CubeCLT installation!")
```